modules/adspiritBidAdapter.js

```diff
@@ -0,0 +1,137 @@
+  aliases: ['twiago'],
+  supportedMediaTypes: [BANNER, NATIVE],
+
+  isBidRequestValid: function (bid) {
+    let host = spec.getBidderHost(bid);
+    if (!host || !bid.params.placementId) {
+      return false;
+    }
+    return true;
+  },
+
+  buildRequests: function (validBidRequests, bidderRequest) {
+    let requests = [];
+    for (let i = 0; i < validBidRequests.length; i++) {
+      let bidRequest = validBidRequests[i];
+      bidRequest.adspiritConId = spec.genAdConId(bidRequest);
+      let reqUrl = spec.getBidderHost(bidRequest);
+      let placementId = utils.getBidIdParameter('placementId', bidRequest.params);
+      reqUrl = '//' + reqUrl + RTB_URL + '&pid=' + placementId +
+        '&ref=' + encodeURIComponent(bidderRequest.refererInfo.topmostLocation) +
+        '&scx=' + (screen.width) +
+        '&scy=' + (screen.height) +
+        '&wcx=' + (window.innerWidth || document.documentElement.clientWidth) +
+        '&wcy=' + (window.innerHeight || document.documentElement.clientHeight) +
+        '&async=' + bidRequest.adspiritConId +
+        '&t=' + Math.round(Math.random() * 100000);
+
+      let data = {};
+
+      if (bidderRequest && bidderRequest.gdprConsent) {
+        const gdprConsentString = bidderRequest.gdprConsent.consentString;
+        reqUrl += '&gdpr=' + encodeURIComponent(gdprConsentString);
+      }
+
+      if (bidRequest.schain && bidderRequest.schain) {
+        data.schain = bidRequest.schain;
+      }
+
+      requests.push({
+        method: 'GET',
+        url: reqUrl,
+        data: data,
+        bidRequest: bidRequest
+      });
+    }
+    return requests;
+  },
+
+  interpretResponse: function (serverResponse, bidRequest) {
+    const bidResponses = [];
+    let bidObj = bidRequest.bidRequest;
+
+    if (!serverResponse || !serverResponse.body || !bidObj) {
+      utils.logWarn(`No valid bids from ${spec.code} bidder!`);
+      return [];
+    }
+
+    let adData = serverResponse.body;
+    let cpm = adData.cpm;
+
+    if (!cpm) {
+      return [];
+    }
+
+    let host = spec.getBidderHost(bidObj);
+    let bidResponse;
+    if ('mediaTypes' in bidObj && 'native' in bidObj.mediaTypes) {
+      bidResponse = {
+        requestId: bidObj.bidId,
+        cpm: cpm,
+        width: adData.w,
+        height: adData.h,
+        creativeId: bidObj.params.placementId,
+        currency: 'EUR',
+        netRevenue: true,
+        ttl: 300,
+        meta: {
+          advertiserDomains: bidObj && bidObj.adomain ? bidObj.adomain : []
+        },
+        native: {
+          title: adData.title,
+          body: adData.body,
+          cta: adData.cta,
+          image: { url: adData.image },
+          clickUrl: adData.click,
+          impressionTrackers: [adData.view]
+        },
+        mediaType: NATIVE
+      };
+    } else {
+      let adm = '<script>window.inDapIF=false</script><script src="//' + host + SCRIPT_URL + '"></script><ins id="' + bidObj.adspiritConId + '"></ins>' + adData.adm;

```

Comment:
We do not allow the loading of outside scripts. Please update this

Reply:
whats the difference if the server responds with a script or if the code creates the script?